python/GafferUI/NameValuePlugValueWidget.py

```diff
@@ -51,14 +53,25 @@ class NameValuePlugValueWidget( GafferUI.PlugValueWidget ) :
 
 	def __init__( self, childPlug ) :
 
+		if isinstance( childPlug, Gaffer.Plug ) :
+			childPlugs = { childPlug }
+		else :
+			childPlugs = childPlug or set()
+
 		self.__row = GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Horizontal, spacing = 4 )
 
-		GafferUI.PlugValueWidget.__init__( self, self.__row, childPlug )
+		GafferUI.PlugValueWidget.__init__( self, self.__row, childPlugs )
+
+		assert( len( childPlugs ) > 0 )
```

Comment:
This isn't really a legitimate assertion from the point of view of PlugValueWidget. The idea is that you should be able to construct a widget without a plug and provide one later. We'd need to move a lot of the constructor logic into `setPlugs()` to do that properly, and I think it's OK to avoid that churn for now. Just wanted to note for later that this assertion is more a warning about current limitations than an assertion about reasonable input. Maybe worth a comment?

Reply:
Added [todo](https://github.com/GafferHQ/gaffer/pull/3980/commits/749c13313100b5139ff766cb0e8fbd7a807f4fb9).